WebDriverAgentLib/Utilities/FBConfiguration.m

```diff
@@ -411,4 +436,22 @@ + (BOOL)reduceMotionEnabled
   }
   return NO;
 }
+
+/**
+ Returns defaul parameters for traversing elements tree from parents to children while requesting XCElementSnapshot.
+ The original maxDepth parameter in a dictionary provided by XCAXClient_iOS's method defaultParams is set to NSIntegerMax.
+ This method returns parameters with maxDepth set to FBDefaultSnapshotRequestMaxDepth
+*/
++ (NSMutableDictionary *)defaultSnapshotRequestParameters
```

Comment:
should this dictionary be mutable?

Reply:
well. there are two options. either ```setSnapshotMaxDepth``` replaces the whole dictionary that is returned by ```snapshotRequestParameters``` (then the ```@{}``` syntax will be available), whether the ```setSnapshotMaxDepth``` only changes a one parameter in a mutable dictionary. :-)

Reply:
makes sense. We could also use setValue:forKey: on an immutable dict as a possible workaround

Reply:
let me try

Reply:
Well, left with ```NSMutableDictionary```. anyway the ```snapshotRequestParameters``` returns pointer of type ```NSDictionary```. should not be a problem here. ```setValue:forKey:``` didn't work for me on ```NSDictionary```.